### PR TITLE
Add open boundaries [!pr]

### PR DESCRIPTION
**Pull request by brachbach**
A version of https://github.com/oughtinc/ergo/pull/352 that's built on more recent code and that we now use in the Elicit open-boundaries PR

### CODE REVIEW

File: ergo/distributions/histogram.py [hunk 1]
```diff
@@ -77,8 +77,59 @@ def initialize_optimizable_params(fixed_params):
         num_bins = fixed_params.get("num_bins", 100)
         return onp.full(num_bins, -float(num_bins))
 
-    def normalize(self):
-        return HistogramDist(self.logps, scale=Scale(0, 1))
+    def normalize(self, true_scale: Scale = None):
```
Comment (stuhlmueller) [inline ergo/distributions/histogram.py:80]: Feels like this should be split into two functions:

1. Expand the scale and add empty bins
2. Normalize

Is there a reason to merge these?

File: ergo/distributions/histogram.py [hunk 2]
```diff
@@ -182,14 +233,29 @@ def to_lists(self, true_scale=True, verbose=False):
 
         return xs, ps
 
-    def to_pairs(
-        self, true_scale=True, verbose=False,
-    ):
+    def to_pairs(self, true_scale=True, verbose=False, total_p: float = 1.0):
```
Comment (stuhlmueller) [inline ergo/distributions/histogram.py:236]: Doesn't feel right to add this transformation here. Options that seem better:

1. Make a separate function that rescales the pairs
2. Support histograms that don't sum to 1 more generally, add a rescale method to HistogramDist that returns a rescaled histogram, then just export using the old `to_pairs` function

File: ergo/distributions/histogram.py (hunk 1 above)
Comment (brachbach) [reply ergo/distributions/histogram.py:80]: nope!

File: ergo/distributions/histogram.py (hunk 2 above)
Comment (brachbach) [reply ergo/distributions/histogram.py:236]: 1 seems good, I think 2 seems overengineered

**Comment by brachbach** [comment]
@stuhlmueller I agree with your style comments, but if I don't end up having to touch Ergo again to improve Elicit, I may just leave this stuff as-is for now and do a follow-on PR to refactor

**Pull request closed without merging**